src/test/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParamsTest.java

```diff
@@ -170,4 +173,12 @@ public void shouldApplyToSettledDateCorrectly() {
         assertThat(transactionSearchParams.getQueryMap().get("to_settled_date"), is(ZonedDateTime.parse("2020-09-27T00:00:00.000Z")));
         assertThat(transactionSearchParams.buildQueryParamString(1L), containsString("to_settled_date=2020-09-26"));
     }
-}
\ No newline at end of file
+
+    @ParameterizedTest
+    @CsvSource({"?, %3F", "{ , %7B", "[ , %5B", "f{o{o}{ , f%7Bo%7Bo%7D%7B", "foo&, foo&", "foo@ , foo%40", "foo@@ , foo%40%40", "foo=bar&baz=quux , foo=bar&baz=quux", "foo bar , foo+bar"})
+    public void shouldUrlEncodeOnlyPrescribedSpecialCharactersOnly(String key, String value){
+        transactionSearchParams.setReference(key);
+        String expectedSelfLink = String.format("reference=%s&page=1&display_size=500", value);
```

Comment:
I almost missed the `%s` here: I string concatenation would actually be clearly for this specific line.